doc/map.md

```diff
@@ -0,0 +1,541 @@
+        [1] synchronized: bool
+
+- DDNet is at `version` = 2, Vanilla chooses 3 for all envelopes when one of them uses a bezier curve, but falls back to 2 when there is none.
+- `channel` holds the type of the envelope
+    - 1 -> Sound envelope
+    - 3 -> Position envelope
+    - 4 -> Color envelope
+- `synchronized` has the effect that the envelope syncs to server time, not player join time
+- `start_point` is the index of its first envelope point
+- `num_points` is the number of envelope points for this envelope
+
+See Envelope Points to see how the envelope points are stored.
+
+Envelope Points
+---------------
+
+- `type_id` = 6
+- exactly one item
+
+The `item_data` of the only item contains all the envelope points used for the envelopes.
+
+- Size of each envelope point:
+    - 22 i32s, if all envelopes have `version` = 3
+    - 6 i32s, if all envelopes have a `version` <= 2
+- Note that all unused fields are zeroed
+
+The first 6 i32 of each envelope point, depending on the envelope type it belongs to:
+
+    sound envelope point:
+        [1] time
+        [1] curve type
+        [1] volume
+        [3] -
+
+    position envelope point:
+        [1] time
+        [1] curve_type
+        [1] x
+        [1] y
+        [1] rotation
+        [1] -
+    
+    color envelope point:
+        [1] time
+        [1] curve type
+        [4] color: I32Color
+
+- `time` is the timestamp of the point, it should increase monotonously within each envelope
+- `curve_type` holds how the curve should bend between this point and the next one
+    - 0 -> Step (abrupt drop at second value)
+    - 1 -> Linear (linear value change)
+    - 2 -> Slow (first slow, later much faster value change)
+    - 3 -> Fast (first fast, later much slower value change)
+    - 4 -> Smooth (slow, faster, then once more slow value change)
+    - 5 -> Bezier (Vanilla only, very customizable curve)
+
+- `x` and `y` hold the movement
+- **I32Color** actually means that the color values for r, g, b, a are i32 values
+
+If bezier curves are used anywhere (envelope version 3), then there are 16 more i32 for each point.
+These are only non-zero if the `curve_type` of the point is 5 (Bezier):
+
+    bezier point extension:
+        [4] in_tangent_dx
+        [4] in_tangent_dy
+        [4] out_tangent_dx
+        [4] out_tangent_dy
+
+Groups
+------
+
+- `type_id` = 4
+
+
+    item_data of group items
+        [1] version
+        [1] x_offset
+        [1] y_offset
+        [1] x_parallax
+        [1] y_parallax
+        [1] start_layer
+        [1] num_layers
+        
+        version 2 extension:
+        [1] clipping: bool
+        [1] clip_x
+        [1] clip_y
+        [1] clip_width
+        [1] clip_height
+        
+        version 3 extension:
+        [3] name: I32String
+        
+- both Vanilla and DDNet are at `version` = 3
+- `start_layer` and `num_layers` tell you which layers belong to this group. Obviously groups are not allowed to overlap
```

Comment:
But the reference implementation does not enforce it at the time of writing?

Reply:
I don't believe that this is the case. It theoretically also allows reuse of envelope points for multiple envelopes if I'm not mistaken.
I do believe however that this should be a rule, since i see no use other than zip-bombing clients, by having multiple small items load the same data-item. Disallowing reuse of data-items should be a general rule imo.